From 9578f18c649dea0bc75b3f262403bdcdd7c9b332 Mon Sep 17 00:00:00 2001
From: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Date: Tue, 1 Dec 2015 15:50:14 +0530
Subject: [PATCH 0202/1851] media: xilinx: Make use of MEDIA_BUS_FMT
 definitions

commit 299bfeff641fd4bdeec9eb623f77af8339016b7c from
https://github.com/Xilinx/linux-xlnx.git

Since IP core doesn't touch memory base the
format decision on xvip_format->code instead
of fourcc.

Signed-off-by: Radhey Shyam Pandey <radhey.shyam.pandey@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Tested-by: Christian Kohn <christian.kohn@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-tpg.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-tpg.c b/drivers/media/platform/xilinx/xilinx-tpg.c
index 6e87272ef9e0..4e8c1718b601 100644
--- a/drivers/media/platform/xilinx/xilinx-tpg.c
+++ b/drivers/media/platform/xilinx/xilinx-tpg.c
@@ -223,14 +223,14 @@ static int xtpg_s_stream(struct v4l2_subdev *subdev, int enable)
 	if (xtpg->is_hls) {
 		u32 fmt;
 
-		switch (xtpg->vip_format->fourcc) {
-		case V4L2_PIX_FMT_YUYV:
+		switch (xtpg->vip_format->code) {
+		case MEDIA_BUS_FMT_UYVY8_1X16:
 			fmt = XTPG_HLS_COLOR_FORMAT_YUV_422;
 			break;
-		case V4L2_PIX_FMT_YUV444:
+		case MEDIA_BUS_FMT_VUY8_1X24:
 			fmt = XTPG_HLS_COLOR_FORMAT_YUV_444;
 			break;
-		case V4L2_PIX_FMT_RGB24:
+		case MEDIA_BUS_FMT_RBG888_1X24:
 			fmt = XTPG_HLS_COLOR_FORMAT_RGB;
 			break;
 		}
-- 
2.31.1

